packages/components/src/internal/components/domainproperties/BasePropertiesPanel.tsx

```diff
@@ -96,6 +98,11 @@ export class BasePropertiesPanel extends React.PureComponent<Props, any> {
                         <Alert bsStyle="danger">{PROPERTIES_PANEL_ERROR_MSG}</Alert>
                     </div>
                 )}
+                {isValid && hasWarning && (
+                    <div onClick={this.toggleLocalPanel} className={getDomainAlertClasses(collapsed, true, useTheme)}>
+                        <Alert bsStyle="warning">{PROPERTIES_PANEL_NAMING_PATTERN_WARNING_MSG}</Alert>
+                    </div>
+                )}
```

Comment:
Since the `PROPERTIES_PANEL_NAMING_PATTERN_WARNING_MSG` seems rather specific to that usage of `BasePropertiesPanel`, would it make sense to instead pass that as a `warning` string property and use that instead of `hasWarning`?

```suggestion
                {isValid && warning && (
                    <div onClick={this.toggleLocalPanel} className={getDomainAlertClasses(collapsed, true, useTheme)}>
                        <Alert bsStyle="warning">{warning}</Alert>
                    </div>
                )}
```

Reply:
Although I don't have a huge preference, it seems a little cleaner to me in this case to pass a boolean within `DataClassPropertiesPanel` and `SampleTypePropertiesPanel` to `BasePropertiesPanel`, and also down to `EntityDetailsForm` where the bool is used in a `classNames`, than to pass a null or a string—although I would agree in other cases that it's better to collapse the two and use a `warning`-style pattern. Further thoughts welcome, though!

Reply:
(I ended up actually going for this and having the prop pass down either undefined or a warning message string, since it was easier to include the prefix with the message this way!)